mdata/aggmetric.go

```diff
@@ -446,6 +448,19 @@ func (a *AggMetric) Add(ts uint32, val float64) {
 		return
 	}
 
+	// need to check if ts > futureTolerance to prevent that we reject a datapoint
+	// because the ts value has wrapped around the uint32 boundary
+	if ts > a.futureTolerance && int64(ts-a.futureTolerance) > time.Now().Unix() {
+		discardedSampleTooFarAhead.Inc()
```

Comment:
this should only be incremented if we discard the sample

Reply:
dieter and i decided to do it this way because we want to be able to predict whether data points would get rejected once we enable that feature. that's why the counter gets increased even when enforcement is disabled.

Reply:
The metric is for the number of points that are discarded.  If no point is being discarded, it should not be incremented.  At a minimum you should add a comment to the code explaining why this metric is incremented even if no point is discarded.   It might be better to use a different metric to track the number of points that exceed the futureTolerance but are not discarded.

Reply:
I added a comment to the doc: https://github.com/grafana/metrictank/pull/1572/files#diff-683a405c5c005542f9e848b827fe0765R373
but i can add one more to the code. or should i just rename the metric to something like `tank.sample-too-far-ahead` instead of `tank.discarded.sample-too-far-ahead`?

Reply:
I would prefer to have 2 metrics.  `discardedSampleTooFarAhead` that increments only when points are discarded, and a  `sampleTooFarAhead` that increments when points are too far ahead regardless of whether enforce-future-tolerance is set to true or not.

Reply:
yep, incrementing a counter that says discarded, when we don't discard, is just creating confusion for no good reason.

Reply:
i added a second metric